app/controllers/api/v0/versions_controller.rb

```diff
@@ -25,6 +25,43 @@ def show
     }
   end
 
+  def raw
+    @version ||= version_collection.find(params[:id])
+
+    expires_in 1.year, public: true
+
+    if Archiver.external_archive_url?(@version.uri)
+      redirect_to @version.uri, status: 301
+    elsif Archiver.store.contains_url?(@version.uri)
+      # Get the file
+      path = URI.parse(@version.uri).path
+      filename = File.basename(path)
+      upstream = Archiver.store.get_file(filename)
```

Comment:
I've taken a swing at adding url handling to get_file.  I updated both local and s3 file handling. Not sure if I handled it the right way, but I see what you're saying about leveraging the existing logic inside of file storage to figure out what to do with urls. Maybe it would be better to split that stuff out into a `get_file_from_url` method instead of handling two different types of arguments in `get_file`? Very open to feedback on that.

Reply:
the more I think about my 1st cut on this the more I don't like it, since I've added complexity to `get_file` where the raw route is the only thing using it. Also, tests that are passing locally are failing in circle ci. Maybe I'll go ahead and split that out on the next pass.

Reply:
Ahhhh, that does seem complex. FWIW, I was thinking something like:

```rb
def get_file(path)
  # If it's a URL, get the path
  path = path[7..-1] if path.starts_with? 'file://'

  # If it's absolute, make sure it's in this storage's directory
  absolute = if path.starts_with?('/')
    raise ArgumentError, "The path '#{path}' does not belong to this storage object" unless path.starts_with?(File.join(directory, ''))
    path
  else
    full_path(path)
  end

  File.read(full_path(path))
end

…but not sure if I’m missing some major issue here.

Reply:
Update: fixed a typo above!

Reply:
Ok, let’s not delay this for something that seems like it might be introducing confusion or complexity. Why don’t you go ahead and pull the `get_file` changes back out and we can probably call this PR ready to go?

Reply:
Went ahead and reverted that commit.